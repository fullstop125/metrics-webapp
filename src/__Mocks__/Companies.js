import React from 'react';

const Companies = () => (
  <>
    <div className="search">
      <input
        className=" search-input"
        type="text"
        onChange={(e) => e.target.value}
        placeholder="Search Company"
      />
      <select className="drop-down" onChange={(e) => e.target.value}>
        <option value="">Select Company</option>

        <option>name</option>
      </select>
    </div>
    <div className="container">
      <h3>Industrial Average</h3>
    </div>
    <div className="bg-pink">
      <h3 className="company-name">Companies Name</h3>
      <div className="row">
        companiesData
        <div className="card" aria-hidden="true">
          <ul className="lists">
            <li>
              name
              <span>symbol</span>
            </li>
            <li>headQuarter</li>
          </ul>
        </div>
      </div>
    </div>
  </>
);

export default Companies;
