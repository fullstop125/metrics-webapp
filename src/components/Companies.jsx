import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FaArrowCircleRight } from 'react-icons/fa';
import { v4 as uuidv4 } from 'uuid';
import { useNavigate } from 'react-router-dom';
import { fetchData } from '../redux/companies/companies';

const Companies = () => {
  const navigate = useNavigate();
  const { companiesData } = useSelector((state) => state.companies);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchData());
  }, []);

  return (
    <>
      <div className="container">
        <h3>Industrial Average</h3>
      </div>
      <div className="bg-pink">
        <h3 className="company-name">Companies Name</h3>
        <div className="row">
          {companiesData.map((company) => (
            <div
              key={uuidv4()}
              className="card"
              onClick={() => navigate(`/details/${company.symbol}`)}
              aria-hidden="true"
            >
              <FaArrowCircleRight className="icon-arrow" />
              <ul className="lists">
                <li>
                  {company.name}
                  <span>
                    (
                    {company.symbol}
                    )
                  </span>
                </li>
                <li>{company.headQuarter}</li>
              </ul>
            </div>
          ))}
        </div>
      </div>
    </>
  );
};

export default Companies;
