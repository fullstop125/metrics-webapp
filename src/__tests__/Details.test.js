import React from 'react';
import renderer from 'react-test-renderer';
import Companies from '../__Mocks__/Companies';

describe('Companies', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Companies />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
