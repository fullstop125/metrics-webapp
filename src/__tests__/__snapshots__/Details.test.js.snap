// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies renders correctly 1`] = `
[
  <div
    className="search"
  >
    <input
      className=" search-input"
      onChange={[Function]}
      placeholder="Search Company"
      type="text"
    />
    <select
      className="drop-down"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select Company
      </option>
      <option>
        name
      </option>
    </select>
  </div>,
  <div
    className="container"
  >
    <h3>
      Industrial Average
    </h3>
  </div>,
  <div
    className="bg-pink"
  >
    <h3
      className="company-name"
    >
      Companies Name
    </h3>
    <div
      className="row"
    >
      companiesData
      <div
        aria-hidden="true"
        className="card"
      >
        <ul
          className="lists"
        >
          <li>
            name
            <span>
              symbol
            </span>
          </li>
          <li>
            headQuarter
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;
